File: app/src/main/java/com/kelsos/mbrc/mappers/InetAddressMapper.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.data.ConnectionSettings;
import java.net.InetSocketAddress;
import java.net.SocketAddress;

public class InetAddressMapper implements Mapper<ConnectionSettings, SocketAddress> {
  @Override
  public SocketAddress map(ConnectionSettings connectionSettings) {
    return new InetSocketAddress(connectionSettings.getAddress(), connectionSettings.getPort());
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/InetAddressMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.data.ConnectionSettings
import java.net.InetSocketAddress
import java.net.SocketAddress

class InetAddressMapper : Mapper<ConnectionSettings, SocketAddress> {
  override fun map(connectionSettings: ConnectionSettings): SocketAddress {
    return InetSocketAddress(connectionSettings.address, connectionSettings.port)
  }
}
